config/initializers/rack_cors.rb

```diff
@@ -0,0 +1,25 @@
+## Configure Rack CORS Middleware, so that CloudFront can serve our assets.
+## See https://github.com/cyu/rack-cors
+## And see https://stackoverflow.com/a/36585871
+if defined? Rack::Cors
+  Rails.configuration.middleware.insert_before 0, Rack::Cors do
+    allow do
+      origins [
+        # SERVER_DOMAIN should be set to the current env root web address
+        # such as https://staging.idseq.net/
+        ENV["SERVER_DOMAIN"] || "",
```

Comment:
These configurations will also be applied to the API itself, so I'm wondering if we should configure these things based on the environment.

You could do something like:

```diff
# config/environments/development.rb
+ config.allowed_cors_origins = ["http://localhost:3000", "http://127.0.0.1:3000/"]

# config/environments/staging.rb
+ config.allowed_cors_origins = ["https://staging.idseq.net", "https://www.staging.idseq.net", "https://assets.staging.idseq.net"]
```

And here in `rack_cors.rb` you could use:
```ruby
Rails.configuration.middleware.insert_before 0, Rack::Cors do
    allow { Rails.application.config.allowed_cors_origins }
end
```